python/metrics/src/hydrotools/metrics/flow_duration_curve.py

```diff
@@ -0,0 +1,390 @@
+        probabilities. These values usually represent the "y-values" on a flow
+        duration curve plot.
+    """
+    # Validate
+    size = len(series)
+    assert size > 0, "series is empty"
+    assert 0.0 <= alpha <= 1.0, "alpha must be between [0.0, 1.0]"
+    assert 0.0 <= beta <= 1.0, "beta must be between [0.0, 1.0]"
+
+    # Compute
+    return (
+        (np.arange(size) + 1 - alpha) / (size + 1 - alpha - beta),
+        np.sort(series)[::-1]
+        )
+
+def pearson_flow_duration_curve(
+        series: npt.ArrayLike,
+        number_of_points: int | None = None
+        ) -> tuple[npt.NDArray, npt.NDArray]:
+    """
+    Generate a flow duration curve (FDC) by fitting an array of streamflow
+    values to a log-pearson type III distribution. The series values determine
+    the type of FDC returned. A series of daily or hourly streamflow values
+    will result in a "period-of-record" FDC. A series of annual maximum flows
+    will result in an annual peak FDC, from which you can derive "annual
+    recurrence intervals."
+
+    Parameters
+    ----------
+    series: array-like, required
+        An array of numerical streamflow values used to fit a flow duration
+        curve.
+    number_of_points: int, optional
+        Number of points in the resulting FDC. This directly determines the
+        length of the returned probabilities numpy.ndarray. Point will be
+        regularly sampled from the minimum value in series to the maximum
+        value in series. The default `number_of_points` is equal to the length
+        of series.
+    
+    Returns
+    -------
+    probabilities: numpy.ndarray
+        Array of values that represent exceedance probabilities, also called
+        exceedance frequencies. The range of possible values is [0.0, 1.0].
+        These values usually represent the "x-values" on a flow duration curve
+        plot. These values can be converted into recurrence intervals by
+        dividing them into 1.0 (i.e. 1.0 / probabilities). Note that these
+        recurrence intervals will reflect the sampling frequency of the input
+        series (e.g. "hourly", "daily", or "annual" recurrence intervals). This
+        array will have the same length as series.
+    values: numpy.ndarray
+        Array of actual streamflow values in the same units as the input
+        series. These values correspond directly to exceedance frequencies in
+        probabilities. These values usually represent the "y-values" on a flow
+        duration curve plot.
+    """
+    # Validate
+    size = len(series)
+    assert size > 0, "series is empty"
+    assert np.all(series > 0.0), "log pearson cannot fit zero values"
+    if number_of_points is not None:
+        assert number_of_points > 0, "number_of_points must be positive"
+    else:
+        number_of_points = len(series)
+
+    # Fit
+    log_series = np.log(series)
+    parameters = pearson3.fit(log_series)
```

Comment:
SciPy statistics classes use Maximum Likelihood Estimation which is a perfectly suitable method for fitting the data.  Some literature prefers using Linear Moment Estimation for this purpose as it tends to be more robust and less sensitive to outliers, (checkout [lmoments3](https://github.com/OpenHydrology/lmoments3) which would add another dependency which may not be ideal, and [scipy's lmoments](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.lmoment.html)).  Regardless I would specify that the data is fit with Maximum Likelihood Estimation in the docstring.